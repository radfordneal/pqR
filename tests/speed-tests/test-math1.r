source("time.r")
 
f <- test.cmp(function(n){ for (i in 1:n) a <- sqrt(b); a })

g <- test.cmp(function(n){ for (i in 1:n) a <- sin(b); a })

h <- test.cmp(function(n){ for (i in 1:n) a <- lgamma(b); a })

test.name <- "math1.scalar"

b <- 0.7

sys.time(a<-f(3000000))
print(a)

sys.time(a<-g(3000000))
print(a)

sys.time(a<-h(2000000))
print(a)

test.name <- "math1.vec50"

b <- seq(0.1,0.7,length=50)

sys.time(a<-f(900000))
print(a[c(1,50)])

sys.time(a<-g(500000))
print(a[c(1,50)])

sys.time(a<-h(100000))
print(a[c(1,50)])

test.name <- "math1.vec1000"

b <- seq(0.1,0.7,length=1000)

sys.time(a<-f(150000))
print(a[c(1,1000)])

sys.time(a<-g(70000))
print(a[c(1,1000)])

sys.time(a<-h(10000))
print(a[c(1,1000)])
