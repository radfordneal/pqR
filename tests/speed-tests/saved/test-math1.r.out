
---- math1.scalar 

Time:
[1] 0.83666
Time:
[1] 0.6442177
Time:
[1] 0.2608672

---- math1.vec50 

Time:
[1] 0.3162278 0.8366600
Time:
[1] 0.09983342 0.64421769
Time:
[1] 2.2527127 0.2608672

---- math1.vec1000 

Time:
[1] 0.3162278 0.8366600
Time:
[1] 0.09983342 0.64421769
Time:
[1] 2.2527127 0.2608672
